internal/provider/resource_certificate.go

```diff
@@ -0,0 +1,148 @@
+		},
+		Schema: map[string]*schema.Schema{
+			"csr_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "The ID of the CSR generated for a resource",
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"ca_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Certificate of the CA issuing the replacement certificate",
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"certificate": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "The resulting Certificate details",
+				Elem:        certificates.CertificateSchema(),
+			},
+		},
+	}
+}
+
+func resourceResourceCertificateCreate(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcfClient := meta.(*api_client.SddcManagerClient)
+	apiClient := vcfClient.ApiClient
+
+	csrID := data.Get("csr_id").(string)
+	csrIdComponents := strings.Split(csrID, ":")
+	if len(csrIdComponents) != 4 {
+		return diag.FromErr(fmt.Errorf("CSR ID invalid"))
+	}
+
+	domainID := csrIdComponents[1]
+	resourceType := csrIdComponents[2]
+	caType := data.Get("ca_id").(string)
+
+	resourceFqdn, err := certificates.GetFqdnOfResourceTypeInDomain(ctx, domainID, resourceType, apiClient)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if resourceFqdn == nil {
+		return diag.FromErr(fmt.Errorf("could not determine FQDN for resourceType %s in domain %s", resourceType, domainID))
+	}
+
+	err = certificates.GenerateCertificateForResource(ctx, vcfClient, &domainID, &resourceType, resourceFqdn, &caType)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	certificateOperationSpec := &models.CertificateOperationSpec{
+		OperationType: resource_utils.ToStringPointer("INSTALL"),
+		Resources: []*models.Resource{{
+			Fqdn: *resourceFqdn,
+			Type: &resourceType,
+		}},
+	}
+	replaceCertificatesParams := certificatesSdk.NewReplaceCertificatesParamsWithContext(ctx).
+		WithTimeout(constants.DefaultVcfApiCallTimeout).
+		WithDomainName(domainID)
+	replaceCertificatesParams.SetCertificateOperationSpec(certificateOperationSpec)
+
+	var taskId string
+	responseOk, responseAccepted, err := apiClient.Certificates.ReplaceCertificates(replaceCertificatesParams)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if responseOk != nil {
+		taskId = responseOk.Payload.ID
+	}
+	if responseAccepted != nil {
+		taskId = responseAccepted.Payload.ID
+	}
+	err = vcfClient.WaitForTaskComplete(ctx, taskId, true)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	data.SetId("cert:" + domainID + ":" + resourceType + ":" + taskId)
+
+	return resourceResourceCertificateRead(ctx, data, meta)
+}
+
+func resourceResourceCertificateRead(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcfClient := meta.(*api_client.SddcManagerClient)
+	apiClient := vcfClient.ApiClient
+
+	csrID := data.Get("csr_id").(string)
+	csrIdComponents := strings.Split(csrID, ":")
```

Comment:
Do we need this validation in the provider? The backend could change the format right?

Was this required because the API does not return the fields independently? Please file a bug for this.

Reply:
This is a sanity validation.  Since the creation of CSR, rotation of certificate are a one-off operations I've decided to form their id with the following formula: :<domain_id>:<resource_type>:<task_id> to uniquely identify the operation instance.